src/components/otpInput/OTPInput.tsx

```diff
@@ -31,105 +33,112 @@ type Props = {
+        setHasError(true);
+        triggerHaptic("notificationError");
+        // eslint-disable-next-line functional/immutable-data
+        translate.value = shakeAnimation();
 
-      if (timerRef.current) {
-        clearTimeout(timerRef.current);
+        if (timerRef.current) {
+          clearTimeout(timerRef.current);
+        }
+        // eslint-disable-next-line functional/immutable-data
+        timerRef.current = setTimeout(() => {
+          setHasError(false);
+          onValueChange("");
+        }, 500);
       }
-      // eslint-disable-next-line functional/immutable-data
-      timerRef.current = setTimeout(() => {
-        setHasError(false);
-        setInputValue("");
-        onValueChange("");
-      }, 500);
-    }
-  };
+    };
 
-  React.useEffect(() => () => clearTimeout(timerRef.current), []);
+    React.useEffect(() => () => clearTimeout(timerRef.current), []);
 
-  const handleChange = (value: string) => {
-    if (value.length > length) {
-      return;
-    }
-    setInputValue(value);
-    onValueChange(value);
-    handleValidate(value);
-  };
+    const handleChange = (value: string) => {
+      if (value.length > length) {
+        return;
+      }
+      onValueChange(value);
+      handleValidate(value);
+    };
 
-  return (
-    <Animated.View style={[{ flexGrow: 1 }, animatedStyle]}>
-      <Pressable
-        onPress={() => {
-          inputRef.current?.focus();
-          setHasFocus(true);
-        }}
-        style={[IOStyles.row, { justifyContent: "space-around" }]}
-        accessible={true}
-        accessibilityLabel="OTP Input"
-        accessibilityValue={{ text: inputValue }}
-      >
-        <TextInput
-          value={inputValue}
-          onChangeText={handleChange}
-          style={{ position: "absolute", opacity: 0 }}
-          maxLength={length}
-          ref={inputRef}
-          onBlur={() => setHasFocus(false)}
-          keyboardType="numeric"
-          inputMode="numeric"
-          returnKeyType="done"
-          textContentType="oneTimeCode"
-          autoComplete={autocomplete ? "sms-otp" : undefined}
+    return (
+      <Animated.View style={[{ flexGrow: 1 }, animatedStyle]}>
+        <Pressable
+          onPress={() => {
+            inputRef.current?.focus();
+            setHasFocus(true);
+          }}
+          ref={ref}
+          style={[IOStyles.row, { justifyContent: "space-around" }]}
           accessible={true}
-        />
-        {[...Array(length)].map((_, i) => (
-          <BoxedInput
-            key={i}
-            status={
-              hasError
-                ? "error"
-                : hasFocus && inputValue.length === i
-                ? "focus"
-                : "default"
-            }
-            secret={secret}
-            value={inputValue[i]}
+          accessibilityLabel={accessibilityLabel}
+          accessibilityHint={accessibilityHint}
+          accessibilityValue={{ text: value }}
+        >
+          <TextInput
```

Comment:
I've found that we are not exposing `InputAccessoryViewID` prop too? We actually use it in app. What do you think?

Reply:
addressed [9f8d73b](https://github.com/pagopa/io-app-design-system/pull/248/commits/9f8d73b3ac64da3c014906893c88d361f4e95925)